virtualmachine/vsphere/util.go

```diff
@@ -1296,5 +1297,11 @@ func updateCustomSpec(vm *VM,
 	gateway := vm.NetworkSettings.Gateway
 	nicSetting.Adapter.Gateway = append(nicSetting.Adapter.Gateway, gateway)
 
+	// set dns server
+	if vm.NetworkSettings.DnsServer != "" {
```

Comment:
Should it append to any dnsServerList in vm-template? I guess here instead it will override same.